web-animations-1/Overview.bs

```diff
@@ -2465,123 +2465,203 @@ Animation effects {#animation-effects}
+  Subsequent levels of this specification
+  will define further types of [=animation effects=].
+
+### Associated Animation and Timeline ### {#animation-effects-and-animations}
+
+  An [=animation effect=] is
+  <dfn for="animation effect" lt="associated animation"
+       local-lt="associated with an animation">associated with an animation</dfn>
+  when the effect is that animation’s [=associated effect=].
   At a given moment,
-  an [=animation effect=] can be [=associated with an animation|associated=]
+  an [=animation effect=] can be [=associated animation|associated=]
   with at most one [=animation=].
 
   An [=animation effect=], |effect|,
   is <dfn>associated with a timeline</dfn>, |timeline|,
   if |effect| is [=associated with an animation=] which, in turn,
   is associated with |timeline|.
 
-### Types of animation effects ### {#types-of-animation-effects}
+### Local time ### {#local-time-section}
 
-  This specification defines a single type of [=animation effect=]:
-  [=keyframe effects=].
-  Subsequent levels of this specification
-  will define further types of [=animation effects=].
+  The <dfn>local time</dfn> of an [=animation effect=]
+  represents the [=animation/current time=]
+  provided by its [=associated animation=],
+  driving the progress of the [=animation effect=].
+  At a given moment,
+  it is based on the first matching condition below:
 
-  All types of [=animation effects=]
-  define a number of common properties
-  which are described in the following sections.
+  <div class=switch>
 
-### The active interval ### {#the-active-interval}
+  : If the [=animation effect=] is [=associated with an animation=],
+  ::
+      the [=local time=] is the [=animation/current time=] of the [=animation=].
+
+  : Otherwise,
+  ::
+      the [=local time=] is [=unresolved=].
+
+  </div>
+
+### Time Spaces ### {#effect-time-spaces}
 
 <div class=informative-bg>
+<em>This section is non-normative</em>
 
-  The period that an [=animation effect=] is scheduled to run
-  is called its [=active interval=].
-  Each [=animation effect=] has only one such interval.
+  In Web Animations all times are relative to some point of reference.
+  These different points of reference produce different <dfn>time spaces</dfn>.
 
-  The lower bound of the [=active interval=] typically corresponds
-  to the [=start time=] of
-  the [=animation=] associated with this [=animation effect=],
-  but can be shifted by a [=start delay=] on the [=animation effect=].
+  This can be compared to coordinate spaces
+  as used in computer graphics.
+  The zero time of a [=time space=] is analogous
+  to the origin of a coordinate space.
 
-  The upper bound of the interval is determined
-  by the [=active duration=].
+  Within the [=Web Animations model=]
+  we define progress of the [=animation effect=] in terms of the [=active time=],
+  which is a time relative to the beginning of the [=active interval=]--
+  thus a time in the [=active time space=].
```

Comment:
Progress is ultimately based on the transformed time space which is closer to iteration time than active time. Can we change this paragraph and the following note to "For example, we can define times relative to the active interval as being in active time space.  This time space, however, is internal to the model and not exposed in the programming interface or in markup." (I don't think the extra highlighting produced by making it a separate "Note:" is warranted here.)

Reply:
How about tweaking the wording to “we define progress of an [=animation effect=] on the basis of its [=active time=]”?

Part of what I was trying to do here was build up a mental model of how these time spaces build on each other. Your example works fine for merely illustrating what a time space is. But doesn't help us understand how the various time spaces build on each other.

Also, keep in mind that the animation effect encompasses the concepts of iteration and iteration direction. So if you lay out the animation effect with all its iterations and time transformations onto a line, progress through the animation effect is progress through all that. It's not progress through the keyframe stack, which is indeed based on the transformed progress / iteration progress.

Reply:
Yeah, my concern is that while "progress" is a general term I tend to equate it with "iteration progress" since that's the real star of the show. Iteration progress is what [`progress`](https://drafts.csswg.org/web-animations-1/#dom-computedeffecttiming-progress) in the API refers to and it's the value that ties the timing model and animation model together.

But in the specific section you're editing (the timing model section), we talk about different kinds of progress like "overall progress" and "transformed progress" so I think what you have makes sense. I think the tweak you proposed here ("...on the basis of...") makes it clearer still. Thanks!